src/cobra/core/metabolite.py

```diff
@@ -40,7 +44,30 @@ class Metabolite(Species):
        Compartment of the metabolite.
     """
 
-    def __init__(self, id=None, formula=None, name="", charge=None, compartment=None):
+    # noinspection PyShadowingBuiltins
+    def __init__(
+        self,
+        id: str = None,
+        formula: str = None,
+        name: str = "",
+        charge: float = None,
+        compartment: str = None,
```

Comment:
Always make type annotations `Optional` when the argument can be `None`, please.